tests/functional/src/main/java/org/mule/functional/junit4/FunctionalTestCase.java

```diff
@@ -267,8 +275,92 @@ protected MuleEvent runFlow(String flowName) throws Exception
+     * @param action See {@link TransactionConfig} constants
+     * @param factory
+     * @return the resulting <code>MuleEvent</code>
+     * @throws Exception
+     */
+    protected MuleEvent runTransactionalFlow(String flowName, byte action, TransactionFactory factory) throws Exception
+    {
+        return runTransactionalFlow(flowName, (Object) null, action, factory);
+    }
+
+    /**
+     * Runs the given flow with a default event inside a transaction
+     * 
+     * @param flowName the name of the flow to be executed
+     * @param payload the payload to use in the message
+     * @param action See {@link TransactionConfig} constants
+     * @param factory
+     * @return the resulting <code>MuleEvent</code>
+     * @throws Exception
+     */
+    protected <T> MuleEvent runTransactionalFlow(String flowName, T payload, byte action, TransactionFactory factory) throws Exception
+    {
+        MuleTransactionConfig transactionConfig = new MuleTransactionConfig(action);
+        transactionConfig.setFactory(new XaTransactionFactory());
+
+        ExecutionTemplate<MuleEvent> executionTemplate = createTransactionalExecutionTemplate(muleContext, transactionConfig);
+
+        return executionTemplate.execute(new ExecutionCallback<MuleEvent>()
+        {
+            @Override
+            public MuleEvent process() throws Exception
+            {
+                return runFlow(flowName, payload);
+            }
+        });
+    }
+
+    /**
+     * Runs the given flow with a default event inside a transaction expecting a failure. Will fail if there's no
+     * failure running the flow.
+     *
+     * @param flowName the name of the flow to be executed
+     * @param action See {@link TransactionConfig} constants
+     * @param factory
+     * @return the message exception return by the flow
+     * @throws Exception
+     */
+    protected MessagingException runTransactionalFlowExpectingException(String flowName, byte action, TransactionFactory factory) throws Exception
+    {
+        return runTransactionalFlowExpectingException(flowName, (Object) null, action, factory);
+    }
+
+    /**
+     * Runs the given flow with a default event inside a transaction expecting a failure. Will fail if there's no
+     * failure running the flow.
+     *
+     * @param flowName the name of the flow to be executed
+     * @param payload the payload to use in the message
+     * @param action See {@link TransactionConfig} constants
+     * @param factory
+     * @return the message exception return by the flow
+     * @throws Exception
+     */
+    protected <T> MessagingException runTransactionalFlowExpectingException(String flowName, T payload, byte action, TransactionFactory factory) throws Exception
+    {
+        try
+        {
+            this.runTransactionalFlow(flowName, payload, action, factory);
+        }
+        catch (MessagingException e)
+        {
+            return e;
+        }
+        catch (Exception e)
```

Comment:
Flow is executed inside a ExecutionCallback#process which throws Exception. Why are you adding this restriction here? Is not better to expect an exception and then check that is of an expected type only when requested?